test/CMakeLists.txt

```diff
@@ -65,7 +65,16 @@ include_directories("${GOOGLETEST_ROOT}/googletest/include/")
 include_directories("${SEQAN3_ROOT}/include/")
 include_directories("${SEQAN3_ROOT}/range-v3/include/")
 include_directories("${SEQAN3_ROOT}/sdsl-lite/include/")
+include_directories("${SEQAN3_ROOT}/cereal/include/")
 
+## Optional dependency handling
+option (NO_CEREAL "Don't use cereal, even if present." OFF)
+
+if (NO_CEREAL)
+    add_definitions (-DSEQAN_WITH_CEREAL=0)
+endif (NO_CEREAL)
```

Comment:
we need to actually test the codepaths our users can see, i.e. with and without cereal; see above

Reply:
The thing is that we need to test all test cases that include files that have serialisation code, not only the test cases that actually use this code (because of include and definition errors). And I suspect more than 90% of our test cases include some files that include some serialisation code so it really affects everything.
Just think about it: anything that uses an alphabet, anything that uses a custom container like `seqan3::concatenated_sequences`, all future data structures like alignments or indexes...

We can think about not making every CI run execute everything, but for the nightlies I want all combinations and for CI I want at least one run without CEREAL just to be sure. Since we only test gcc7 right now, I suggest we just have one run CEREAL and one without.